### PR TITLE
Connatix Bid Adapter: support viewability

### PR DESCRIPTION
**Pull request by rares-mihai-preda**
## Type of change

- [ ] Bugfix
- [X] Feature
- [ ] New bidder adapter  
- [X] Updated bidder adapter: https://github.com/prebid/prebid.github.io/pull/5539 
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes

- [X] Does this change affect user-facing APIs or examples documented on http://prebid.org ? https://github.com/prebid/prebid.github.io/pull/5539
- [ ] Other

## Description of change

Connatix Bid Adapter: support viewability with declared viewability from bid params or detect viewability based on a publisher's container unique identifier

## Other information

Update docs: https://github.com/prebid/prebid.github.io/pull/5539

### CODE REVIEW

**Comment by patmmccann** [comment]
None of this is wise in an adapter, it must be imported, as it is getting repeated around the project. See https://github.com/prebid/Prebid.js/blob/master/libraries/percentInView/percentInView.js

**Comment by patmmccann** [comment]
related https://github.com/prebid/Prebid.js/issues/11963#issuecomment-2228765435

**Comment by rares-mihai-preda** [comment]
@patmmccann Thanks for reviewing this! We updated the pull request.

Could you support us please regarding the tests run on the CircleCI because all tests succeed locally and we cannot replicate the behavior from that specific environment?

**Comment by patmmccann** [comment]
Hmm I'm not sure why this test fails in your pr but not local

**Comment by rares-mihai-preda** [comment]
@patmmccann Thanks for looking into this! Since we're unable to replicate the issue locally and it seems specific to the CircleCI environment, could you provide any insights or suggestions on how we might replicate this setup to debug further? 

Alternatively, if you believe this might be an environment-specific issue, would it be possible to merge the PR and track this as a potential bug to investigate later? We're happy to assist in any way we can.

**Comment by rares-mihai-preda** [comment]
@ChrisHuie  Could this be a similar issue to what was encountered in https://github.com/prebid/Prebid.js/pull/12142? All the unit tests pass locally, but they're failing on CircleCI.

**Comment by rares-mihai-preda** [comment]
@ChrisHuie @patmmccann We attempted to run and resolve the tests that failed on CircleCI by executing the following command:

$ gulp test --path 'test/spec/modules/connatixBidAdapter_spec.js' --browsers Safari

While all the tests passed in Safari, they continue to fail in the CircleCI environment. What steps can we take to address this issue?

**Comment by rares-mihai-preda** [comment]
@patmmccann  @ChrisHuie  We've addressed the issue, and the pipeline is now green. Everything is ready for the next steps.

Please let us know if there’s anything else we need to do or if further adjustments are required.
The docs update is also ready here: https://github.com/prebid/prebid.github.io/pull/5539

Thank you!

File: modules/connatixBidAdapter.js [hunk 1]
```diff
@@ -62,6 +67,76 @@ export function validateVideo(mediaTypes) {
...
+export function _getMinSize(sizes) {
+  if (!sizes || sizes.length === 0) return undefined;
+  return sizes.reduce((minSize, currentSize) => {
+    const minArea = minSize.w * minSize.h;
+    const currentArea = currentSize.w * currentSize.h;
+    return currentArea < minArea ? currentSize : minSize;
+  });
+}
+
+export function _isIframe() {
```
Comment (robertrmartinez) [inline modules/connatixBidAdapter.js:79]: There is a function which does this in utils.js please use it

`inIframe`

**Review by robertrmartinez — CHANGES_REQUESTED** [review]
one change please

looks ok

**Comment by rares-mihai-preda** [comment]
@robertrmartinez @patmmccann 
Just following up on our pull request and the refactoring part from August 29. This update is crucial for us, and our clients are eagerly awaiting it. Could you please review it when you have a moment?

Thank you for your time!

**Review by robertrmartinez — APPROVED** [review]

**Comment by rares-mihai-preda** [comment]
@robertrmartinez Thank you for reviewing the pull request!

@ChrisHuie Is there anything else we need to provide to move forward with merging the documentation update for the already merged pull request? Docs update: https://github.com/prebid/prebid.github.io/pull/5539

**Pull request merged**